builder.go

```diff
@@ -95,12 +95,19 @@ func (b *Builder) Build(hs *hschema.HyperSchema) (*types.Root, error) {
 	for i, v := range bs {
 		r.Booleans[i] = v.(*types.Boolean)
 	}
+
 	r.Properties = ps
 
 	for i, l := range hs.Links {
 		var (
 			s, ts types.Schema
+			us    []types.Schema
 		)
+		m, err := GetUrlParameters(l.Href, hs.Schema, ctx)
```

Comment:
Can you put `GetUrlParameters` and `EscapeJSONPath` into one method?
Those methods looks like similar sweeping.